test.rb

```diff
@@ -0,0 +1,155 @@
+
+require 'bolt'
+require 'bolt/project'
+require 'bolt/analytics'
+require 'bolt/executor'
+require 'bolt/config'
+require 'bolt/plugin'
+require 'bolt/inventory'
+require 'bolt/application'
+
+## Steal some forking stuff from puppet based on conversation from https://github.com/puppetlabs/bolt/pull/892/files
+
+# stolen from https://github.com/puppetlabs/puppet/blob/ed44890fa0a85e307d2a15cce20406fa279a70e5/lib/puppet/util.rb#L528
+def safe_posix_fork(stdin = $stdin, stdout = $stdout, stderr = $stderr, &block)
+  Kernel.fork do
+    $stdin.reopen(stdin)
+    $stdout.reopen(stdout)
+    $stderr.reopen(stderr)
+
+    $stdin = STDIN
+    $stdout = STDOUT
+    $stderr = STDERR
+
+    begin
+      Dir.foreach('/proc/self/fd') do |f|
+        if f != '.' && f != '..' && f.to_i >= 3
+          begin
+            IO.new(f.to_i).close
+          rescue StandardError
+            nil
+          end
+        end
+      end
+    rescue Errno::ENOENT, Errno::ENOTDIR # /proc/self/fd not found, /proc/self not a dir
+      3.upto(256) do |fd|
+        IO.new(fd).close
+      rescue StandardError
+        nil
+      end
+    end
+
+    block&.call
+  end
+end
+
+# stolen from https://github.com/puppetlabs/puppet/blob/6b1a1307a0d23a9e24f727c476d4c738b54052ef/lib/puppet/util/execution.rb#L147
+def execute_plan(application, plan_name)
+  begin
+    reader, stdout = IO.pipe
+    stdin = File.open('/dev/null', 'w+')
+    stderr = File.open('/dev/null', 'w+')
+    output = ''
+
+    child_pid = nil
+    begin
+      child_pid = safe_posix_fork(stdin, stdout, stderr) { puts application.run_plan(plan_name, []) }
```

Comment:
For this iteration I simply put the result of the plan run on STDOUT. there is probably a better way to do that?